### ISSUE TITLE
initialHeaderFilter shows the value instead it's name when using boolean value

### ISSUE THREAD

**Issue by RodrigoCL-inkasso** [olifolkerd/tabulator#3901]
Hello. I am having an issue that eventually I solved, but I thought I could share with you guys. It's about the lookup using boolean as values, especially using the initialHeaderFilter property.

I created an example here where I stumbled: https://jsfiddle.net/83ztvsac/15/

I set `initialHeaderFilter:[{field:'gender', value:true}]`

Everything works, but the initial input on the header shows the boolean value, not its representation of the lookup table. It should be showing "Female" at the Gender column's header.

Eventually I solved by converting everything as string, even the data, and using `initialHeaderFilter:[{field:'gender', value:'true'}]`. 

But I think it was worth to report. I'll be looking forward for the fix, because we are migrating the system to use tabulator as the default datagrid.

Thank you very much! 
Rodrigo

**Comment by RodrigoCL-inkasso** [comment]
Oh yeah, and I forgot to mention: the filter doesn't work, for some reason. I commented under the code to show the filter on console. 
If I select, say, "Male", it shows the filter as 
```
[{
  field: "gender",
  type: "=",
  value: "false"
}]
```
but because the values are boolean, not string, this filter doesn't work, and shows nothing. it should be, I think, 
```
[{
  field: "gender",
  type: "=",
  value: false
}]
```
But, then again, it all worked when converted everything as string

**Comment by olifolkerd** [comment]
This is bevause `true` and `false` are not proper values for an object key.

In this case you should be using the array syntax for the values lookup:

```
var genders = [
    {
        label:"Female",
        value:true,
    },
    {
        label:"Male",
        value:false,
    },
    {
        label:"",
        value:"",
    },
]
```

Cheers

Oli :)

**Comment by RodrigoCL-inkasso** [comment]
thanks so much for the response. I'm sorry for insisting.

I edited the Fiddler using this suggestion, and I found out that it still doesn't work, although the header looks ok. Here it is:

[https://jsfiddle.net/18h4du62/](https://jsfiddle.net/18h4du62/)

The header now works, but when I filter for "Male", the filter object array becomes empty, as I captured on the console.log below.

And the columns don't show the "label", only their values. Maybe this array format is not supported by the lookup formatter, nor the "list" filter. 

I am sorry for insisting.

Thanks again.

Rodrigo
